src/display/fetch_stream.js

```diff
@@ -0,0 +1,219 @@
+      let value = this._stream.httpHeaders[property];
+      if (typeof value === 'undefined') {
+        continue;
+      }
+      this._headers.append(property, value);
+    }
+
+    let url = this._stream.source.url;
+    fetch(url, {
+      method: 'GET',
+      headers: this._headers,
+      mode: 'cors',
+      credentials: this._withCredentials ? 'omit' : 'include',
+      redirect: 'follow',
+    }).then((response) => {
+      if (!validateResponseStatus(response.status, this._stream.isHttp)) {
+        throw createResponseStatusError(response.status, url);
+      }
+      this._headersCapability.resolve();
+      this._reader = response.body.getReader();
+
+      let { allowRangeRequests, suggestedLength, } =
+        validateRangeRequestCapabilities({
+          getResponseHeader: (name) => {
+            return response.headers.get(name);
+          },
+          isHttp: this._stream.isHttp,
+          rangeChunkSize: this._rangeChunkSize,
+          disableRange: this._disableRange,
+        });
+
+      this._contentLength = suggestedLength;
+      this._isRangeSupported = allowRangeRequests;
+    }).catch(this._headersCapability.reject);
+
+    this.onProgress = null;
+  }
+
+  get headersReady() {
+    return this._headersCapability.promise;
+  }
+
+  get contentLength() {
+    return this._contentLength;
+  }
+
+  get isRangeSupported() {
+    return this._isRangeSupported;
+  }
+
+  get isStreamingSupported() {
+    return this._isStreamingSupported;
+  }
+
+  read() {
+    return this._headersCapability.promise.then(() => {
+      return this._reader.read().then(({ value, done, }) => {
+        if (done) {
+          return Promise.resolve({ value, done, });
+        }
+        this._loaded += value.byteLength;
+        if (this.onProgress) {
+          this.onProgress({
+            loaded: this._loaded,
+            total: this._contentLength,
+          });
+        }
+        let buffer = new Uint8Array(value).buffer;
+        return Promise.resolve({ value: buffer, done: false, });
+      });
+    });
+  }
+
+  cancel(reason) {
+    if (this._reader) {
+      this._reader.cancel(reason);
+    }
+  }
+}
+
+class PDFFetchStreamRangeReader {
+  constructor(stream, begin, end) {
+    this._stream = stream;
+    this._reader = null;
+    this._loaded = 0;
+
+    this._readCapability = createPromiseCapability();
+    this._isStreamingSupported = !stream.source.disableStream;
+
+    this._headers = new Headers();
+    for (let property in this._stream.httpHeaders) {
+      let value = this._stream.httpHeaders[property];
+      if (typeof value === 'undefined') {
+        continue;
+      }
+      this._headers.append(property, value);
+    }
+
+    let rangeStr = begin + '-' + (end - 1);
+    this._headers.append('Range', 'bytes=' + rangeStr);
+    let url = this._stream.source.url;
+    fetch(url, {
+      method: 'GET',
+      headers: this._headers,
+    }).then((response) => {
```

Comment:
define the same options as in full reader. extract the `createFetchOptions(headers)` method?

Reply:
created `createFetchOptions` with new commit. Thanks.